db/python/layers/project_insights.py

```diff
@@ -301,367 +332,601 @@ async def _stats_es_indexes_query(
+                "projects": project_ids,
+                "sequencing_types": sequencing_types,
             },
         )
-        return sequencing_group_reports
+        families_by_project_id_and_seq_fields = defaultdict(list)
+        for row in families:
+            families_by_project_id_and_seq_fields[
+                (
+                    row["project"],
+                    row["sequencing_type"],
+                    row["sequencing_platform"],
+                    row["sequencing_technology"],
+                )
+            ].append(row)
 
-    async def _get_sequencing_groups_from_analysis_id(self, analysis_id: int):
-        """Get sequencing groups from an analysis id"""
+        return families_by_project_id_and_seq_fields
+    
+    async def _details_stripy_reports(self, project_ids: list[ProjectId]) -> dict[ProjectId, int]:
+        """Get stripy web report links"""
         _query = """
-SELECT
+SELECT 
+    a.project,
+    a.id,
+    a.type,
+    a.timestamp_completed,
+    asg.sequencing_group_id,
+    JSON_EXTRACT(meta, '$.outliers_detected') as outliers_detected,
+    JSON_QUERY(meta, '$.outlier_loci') as outlier_loci
+FROM analysis a
+LEFT JOIN analysis_sequencing_group asg on asg.analysis_id=a.id
+INNER JOIN (
+    SELECT
+        asg.sequencing_group_id,
+        MAX(timestamp_completed) as max_timestamp
+    FROM analysis a
+    LEFT JOIN analysis_sequencing_group asg on asg.analysis_id=a.id
+    WHERE type='web'
+    AND project IN :projects
+    AND JSON_EXTRACT(meta, '$.stage') = 'Stripy'
+    GROUP BY asg.sequencing_group_id
+) max_timestamps ON asg.sequencing_group_id = max_timestamps.sequencing_group_id
+AND a.timestamp_completed = max_timestamps.max_timestamp
+WHERE a.project IN :projects
+AND a.type = 'web'
+AND JSON_EXTRACT(meta, '$.stage') = 'Stripy';
+        """
+        
+        _query_results = await self.connection.fetch_all(
+            _query,
+            {
+                "projects": project_ids,
+            },
+        )
+        return {(row['project'], row['sequencing_group_id']): row for row in _query_results}
+    
+    async def _details_mito_reports(self, project_ids: list[ProjectId]) -> dict[ProjectId, int]:
+        """Get mito web report links"""
+        _query = """
+SELECT 
+    a.project,
+    a.id,
+    a.type,
+    a.timestamp_completed,
     asg.sequencing_group_id
+FROM analysis a
+LEFT JOIN analysis_sequencing_group asg on asg.analysis_id=a.id
+INNER JOIN (
+    SELECT
+        asg.sequencing_group_id,
+        MAX(timestamp_completed) as max_timestamp
+    FROM analysis a
+    LEFT JOIN analysis_sequencing_group asg on asg.analysis_id=a.id
+    WHERE type='web'
+    AND project IN :projects
+    AND JSON_EXTRACT(meta, '$.stage') = 'MitoReport'
+    GROUP BY asg.sequencing_group_id
+) max_timestamps ON asg.sequencing_group_id = max_timestamps.sequencing_group_id
+AND a.timestamp_completed = max_timestamps.max_timestamp
+WHERE a.project IN :projects
```

Comment:
Same treatement as above, but you can probably just ditch the inner query because the other fields are the same